### ISSUE TITLE
telegram-desktop sends a lot of data to dbus and loads CPU

### ISSUE THREAD

**Issue by Galicarnax** [telegramdesktop/tdesktop#16236]
### Steps to reproduce
1. Run the latest version of telegram-desktop

### Expected behaviour
Telegram CPU usage at less than 0.1%

### Actual behaviour
Telegram CPU usage is 4-5%

### Configuration
**Operating system:** Arch Linux (+ `sway` window manager)

**Version of Telegram Desktop:** 2.7.3

**Installation source (Linux Only)**:  distribution package:

Since recently, launching `telegram-desktop` leads to high CPU usage by the following processes:
- `telegram-desktop` (~4%)
- `dbus-daemon` (~3%)
- `systemd-journal` (~1.5%)

The numbers are not terribly high, but enough to make CPU fans in my laptop audible, and that's annoying. Exiting `telegram-desktop` drops the other two processes to normal levels.

Launching `dbus-monitor` and then `telegram-desktop` reveals that dbus is constantly bombarded by messages, here is a sample chunk of the output:
<details><summary><b>dbus-monitor output</b>:</summary>
    
    method call time=1619698266.472379 sender=:1.931380 -> destination=org.freedesktop.DBus serial=2 path=/org/freedesktop/DBus; interface=org.freedesktop.DBus; member=RequestName
       string "org.freedesktop.Notifications"
       uint32 4
    signal time=1619698266.472389 sender=org.freedesktop.DBus -> destination=(null destination) serial=37 path=/org/freedesktop/DBus; interface=org.freedesktop.DBus; member=NameOwnerChanged
       string "org.freedesktop.Notifications"
       string ""
       string ":1.931380"
    method return time=1619698266.472402 sender=org.freedesktop.DBus -> destination=:1.931375 serial=38 reply_serial=29
       uint32 1
    signal time=1619698266.472409 sender=org.freedesktop.DBus -> destination=:1.931380 serial=3 path=/org/freedesktop/DBus; interface=org.freedesktop.DBus; member=NameAcquired
       string "org.freedesktop.Notifications"
    method return time=1619698266.472418 sender=org.freedesktop.DBus -> destination=:1.931380 serial=4 reply_serial=2
       uint32 1
    signal time=1619698266.472426 sender=org.freedesktop.DBus -> destination=:1.931380 serial=10 path=/org/freedesktop/DBus; interface=org.freedesktop.DBus; member=NameLost
       string "org.freedesktop.Notifications"
    signal time=1619698266.472434 sender=org.freedesktop.DBus -> destination=(null destination) serial=39 path=/org/freedesktop/DBus; interface=org.freedesktop.DBus; member=NameOwnerChanged
       string "org.freedesktop.Notifications"
       string ":1.931380"
       string ""
    signal time=1619698266.472447 sender=org.freedesktop.DBus -> destination=:1.931380 serial=11 path=/org/freedesktop/DBus; interface=org.freedesktop.DBus; member=NameLost
       string ":1.931380"
    signal time=1619698266.472455 sender=org.freedesktop.DBus -> destination=(null destination) serial=40 path=/org/freedesktop/DBus; interface=org.freedesktop.DBus; member=NameOwnerChanged
       string ":1.931380"
       string ":1.931380"
       string ""
    error time=1619698266.472467 sender=org.freedesktop.DBus -> destination=:1.931375 error_name=org.freedesktop.DBus.Error.NoReply reply_serial=34
       string "Message recipient disconnected from message bus without replying"
    error time=1619698266.472476 sender=org.freedesktop.DBus -> destination=:1.931375 error_name=org.freedesktop.DBus.Error.NoReply reply_serial=33
       string "Message recipient disconnected from message bus without replying"
    error time=1619698266.472484 sender=org.freedesktop.DBus -> destination=:1.931375 error_name=org.freedesktop.DBus.Error.NoReply reply_serial=32
       string "Message recipient disconnected from message bus without replying"
    method call time=1619698266.472492 sender=:1.931375 -> destination=org.freedesktop.DBus serial=35 path=/org/freedesktop/DBus; interface=org.freedesktop.DBus; member=StartServiceByName
       string "org.freedesktop.Notifications"
       uint32 0
</details>

At the same time, systemd journal prints lots copies of this:
<details><summary><b>systemd journal</b>:</summary>

    Apr 29 18:59:54 <hostname> org.freedesktop.Notifications[2570737]: failed to create display
    Apr 29 18:59:54 <hostname> dbus-daemon[603]: [session uid=1000 pid=603] Activating service name='org.freedesktop.Notifications' requested by ':1.954812' (uid=1000 pid=2570422 comm="telegram-desktop")
    Apr 29 18:59:54 <hostname> dbus-daemon[603]: [session uid=1000 pid=603] Successfully activated service 'org.freedesktop.Notifications'   
</details>

P.S. Just now updated to telegram-desktop v. 2.7.4, the issue is the same

**Comment by ilya-fedin** [comment]
Logs look weird. systemd journal says it's successfully activated, while dbus-monitor says no reply

**Comment by ilya-fedin** [comment]
I can't reproduce that in my environment. I assume something is wrong with your system.

**Comment by IvoAndonov** [comment]
I'm with Arch too and no such problem on my end. Judging by your version, you're using an AUR version of telegram-desktop, so you should take this problem to the AUR maintainer of telegram-desktop-bin. I'm using telegram-desktop from the community repo whose version is 2.7.1 and everything is normal. CPU usage, even at launching the program, is never higher than 0%, I doubt that my CPU even knows I'm running the program. :D 
You also don't give any information about your hardware and if it's an ancient CPU, it's possible it's just too weak to handle the load. Personally, I don't think 5% is something to worry about. I'd be more worried if the program took 30% or more at launching or at simply being minimized in tray.

**Comment by ilya-fedin** [comment]
> whose version is 2.7.1

![изображение](https://user-images.githubusercontent.com/17829319/116634254-15e67c00-a94b-11eb-839c-9a90741877aa.png)

**Comment by Galicarnax** [comment]
> Judging by your version, you're using an AUR version of telegram-desktop

I'm definitely using `telegram-desktop` from the community repo, which is at the moment `v.2.7.4-1`. In the first post I provided version number as declared by Telegram app itself (under the three dashes on the top left), not from `pacman -Qi`. And yes, I also have another machine with Arch Linux at my work, which is runnign `X11/i3` as opposed to `Wayland/sway` on my laptop. And no such problem is observed there with the same version of `telegram-desktop`. Have no idea who is the actual culprit, then.

> You also don't give any information about your hardware

It's Lenovo Legion 5, a gaming laptop, so no problem with CPU power. And this did not happen on this very laptop a few weeks ago, the problem occurred after some of the recent updates.

Is there a way to run `telegram-desktop` in log-enabled debug mode? Didn't find anything relevant in its settings.

**Comment by ilya-fedin** [comment]
> Is there a way to run `telegram-desktop` in log-enabled debug mode?

Debug logs won't help, nothing is logged to them regarding dbus. Generic log is in `~/.local/share/TelegramDesktop/log.txt`.

**Comment by IvoAndonov** [comment]
> I'm definitely using `telegram-desktop` from the community repo, which is at the moment `v.2.7.4-1`. 

At the time of writing my previous post that version hasn't yet arrived on my end. Now I opened Telegram and it said it had updated to 2.7.4. I still don't have your problem, though.
I have found that in such cases the culprit is very hard to find, nearly impossible. The best I can suggest is to make a full backup of your system using a program like Clonezilla and then reinstall Arch with all of the latest packages. That helps sometimes.
